lib/ramble/ramble/test/end_to_end.py

```diff
@@ -800,3 +801,125 @@ def test_custom_executables(mutable_config, mutable_mock_workspace_path, mock_ap
+
+    base_config = """ramble:
+  mpi:
+    command: 'mpirun'
+    args:
+    - '-n'
+    - '{n_ranks}'
+  batch:
+    submit: '{execute_experiment}'
+  applications:\n"""
+
+    list_cmd = RambleCommand('list')
+    info_cmd = RambleCommand('info')
+
+    all_apps = list_cmd().split('\n')
+
+    workload_regex = re.compile(r'Workload: (?P<wl_name>.*)')
+
+    for app_str in all_apps:
+        app = app_str.replace(' ', '')
+        if app != '':
+            print('Application: ' + app)
+            app_info = info_cmd(app)
+            workloads = []
+
+            for line in app_info.split('\n'):
+                match = workload_regex.search(line)
+                if match:
+                    workloads.append(match.group('wl_name').replace(' ', ''))
+
+            ws_name = f'test_all_workloads_{app}'
+            with ramble.workspace.create(ws_name) as ws:
+                ws.write()
+                config_path = os.path.join(ws.config_dir, ramble.workspace.config_file_name)
+
+                with open(config_path, 'w+') as f:
+                    f.write(base_config)
+                    f.write(f'    {app.strip()}:\n')
+                    f.write('      workloads:\n')
+                    for workload in workloads:
+                        f.write(f"""        {workload.strip()}:
+          experiments:
+            test_experiment:
+              variables:
+                n_ranks: '1'
+                n_nodes: '1'
+                processes_per_node: '1'\n""")
+                    f.write("""spack:
+  concretized: false
+  compilers: {}
+  mpi_libraries: {}
+  applications: {}\n""")
+
+                ws._re_read()
+
+                workspace('concretize', global_args=['-w', ws_name])
+                workspace('setup', '--dry-run', global_args=['-w', ws_name])
+
+
+@pytest.mark.long
+def test_dryrun_mock_workloads(mutable_config, mutable_mock_workspace_path, mock_applications):
+
+    base_config = """ramble:
+  mpi:
+    command: 'mpirun'
+    args:
+    - '-n'
+    - '{n_ranks}'
+  batch:
+    submit: '{execute_experiment}'
+  applications:\n"""
+
+    list_cmd = RambleCommand('list')
+    info_cmd = RambleCommand('info')
+
+    all_apps = list_cmd().split('\n')
+
+    workload_regex = re.compile(r'Workload: (?P<wl_name>.*)')
+
+    for app_str in all_apps:
+        app = app_str.replace(' ', '')
+        if app != '':
+            print('Application: ' + app)
+            app_info = info_cmd(app)
+            workloads = []
+
+            for line in app_info.split('\n'):
+                match = workload_regex.search(line)
+                if match:
+                    workloads.append(match.group('wl_name').replace(' ', ''))
+
+            ws_name = f'test_all_workloads_{app}'
+            with ramble.workspace.create(ws_name) as ws:
+                ws.write()
+                config_path = os.path.join(ws.config_dir, ramble.workspace.config_file_name)
```

Comment:
I don't how much of a duplication this is compared to the other test. Is there a way we can make the differences/distinctions more concise? It feels like we should be able to somehow get fancy with the args

Reply:
I parameterized it, so it isn't duplicated. Let me know if you like that more.